packages/rx_bloc_cli/mason_templates/bricks/rx_bloc_base/__brick__/lib/feature_login/ui_components/login_form.dart

```diff
@@ -0,0 +1,132 @@
+// Copyright (c) 2021, Prime Holding JSC
+// https://www.primeholding.com
+//
+// Use of this source code is governed by an MIT-style
+// license that can be found in the LICENSE file or at
+// https://opensource.org/licenses/MIT.
+
+import 'package:flutter/material.dart';
+import 'package:flutter/widgets.dart';
+import 'package:flutter_rx_bloc/flutter_rx_bloc.dart';
+import 'package:flutter_rx_bloc/rx_form.dart';
+import 'package:provider/provider.dart';
+
+import '../../app_extensions.dart';
+import '../../base/common_blocs/user_account_bloc.dart';
+
+class LoginForm extends StatefulWidget {
+  const LoginForm({
+    this.title = 'Enter your login credentials',
+    this.onLoginSuccess,
```

Comment:
I thought that with rx_bloc we don't have to use any StatefulWidget.

Reply:
We should avoid using Stateful widgets as they can cause a lot of side effects and can make widget tests behave unpredictably. However, in some cases they are necessary.

In this particular case we have a list of focus nodes, which if not disposed properly will cause memory leaks. Because of that we should dispose them when not needed (that is, along with the widget). For that we can use the stateful widgets `dispose()` method.